starlette/middleware/sessions.py

```diff
@@ -33,7 +33,7 @@ def __init__(
             await self.app(scope, receive, send)
             return
 
-        request = Request(scope)
+        request = HTTPConnection(scope)
```

Comment:
Yup - we absolutely want this, thanks!
Let's change the variable name to `connection = HTTPConnection(scope)` tho.